lib/src/rules/use_string_buffer.dart

```diff
@@ -0,0 +1,212 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library linter.src.rules.use_string_buffer;
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/token.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/dart/element/element.dart';
+import 'package:linter/src/analyzer.dart';
+import 'package:linter/src/util/dart_type_utilities.dart';
+
+const _desc = r'Use string buffer to compose strings.';
+
+const _details = r'''
+
+**DO** use string buffer to compose strings.
+
+**BAD:**
+```
+String foo() {
+  final buffer = '';
+  for (int i = 0; i < 10; i++) {
+    buffer += 'a'; // LINT
+  }
+  return buffer;
+}
+```
+
+**GOOD:**
+```
+String foo() {
+	final buffer = new StringBuffer();
+  for (int i = 0; i < 10; i++) {
+    buffer.write('a');
+  }
+  return buffer.toString();
+}
+```
+
+''';
+
+SimpleIdentifier _getSimpleIdentifier(AstNode node) {
+  final visitor = new SimpleIdentifierVisitor();
+  node.accept(visitor);
+  return visitor.identifier;
+}
+
+bool _isEmptyInterpolationString(AstNode node) {
+  final visitor = new _IsEmptyInterpolationStringVisitor();
+  node.accept(visitor);
+  return visitor._isEmptyInterpolationString;
```

Comment:
node is InterpolationString makes an implicit cast even with && so it would be the same doing:

```dart
return node is InterpolationString && node.value == '';
```